### ISSUE TITLE
Developers cannot specify optional `dependsOn` steps

### ISSUE THREAD

**Issue by ndowmon** [JupiterOne/sdk#269]
The SDK's logic for `dependsOn` is written such that if a steps dependency is disabled, the step itself becomes disabled.

Developers need more control over `dependsOn` to specify when steps are a dependency iff they are enabled.

An example can be found in the `graph-azure` [project](https://github.com/JupiterOne/graph-azure/blob/master/src/steps/resource-manager/authorization/index.ts#L223). the `STEP_AD_ACCOUNT` is required. On the other hand, the spread `...ROLE_ASSIGNMENT_DEPENDS_ON` steps should be executed prior to this step if they exist, but are not required for this step to execute.

**Comment by austinkelleher** [comment]
Agreed. I think that having more control over our lifecycle functions would make defining our steps a lot more powerful. I had a similar thought with allowing `getStepStartState` to be defined inside of the step itself instead of a single file:

https://github.com/JupiterOne/sdk/issues/82

**Comment by aiwilliams** [comment]
I think we'll need to prioritize this to get the Azure integration working when the instance is configured with AD turned off. @ndowmon would you mind creating an internal issue to track that as the work to do next, with a link back to this? Thanks!
